src/src/components/SelectFormPopper/SelectFormPopper.tsx

```diff
@@ -0,0 +1,173 @@
+// @ts-nocheck
+/* eslint-disable react/prop-types */
+import React from 'react';
+
+import ToggleButton from '@material-ui/lab/ToggleButton';
+import {
+  Checkbox,
+  InputBase,
+  Popper,
+  Snackbar,
+  Tooltip,
+} from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import MuiAlert from '@material-ui/lab/Alert';
+import CheckBoxOutlineBlank from '@material-ui/icons/CheckBoxOutlineBlank';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+
+import { Text } from 'components/kit';
+
+import { ISelectOption } from 'types/services/models/explorer/createAppModel';
+
+const SelectFormPopper: React.FC<ISelectFormPopperProps> = ({
+  id,
+  type,
+  open,
+  anchorEl,
+  options,
+  selectedData,
+  onSelect,
+  searchValue,
+  handleSearchInputChange,
+  handleClose,
+  regexError,
+  setRegexError,
+  isRegexSearch,
+  setIsRegexSearch,
+  classes,
+}) => {
+  return (
+    <Popper
+      id={id}
+      open={open}
+      anchorEl={anchorEl}
+      placement='bottom-start'
+      className={
+        type === 'metrics'
+          ? 'Metrics__SelectForm__Popper'
+          : 'SelectForm__Popper'
```

Comment:
Would it make sense to pass in the className instead of switching on type?

Reply:
There is more than one place where the classes are different that would mean to pass many classNames

Reply:
could you use the same class name in both and customize that CSS in the .scss belonging to each usage?

Reply:
it seems like the "type" switch makes the parent aware of it's children